### PR TITLE
feat: New menu structure 

### LINKED ISSUE #617: New menu structure
### Describe the solution you'd like

The menu should be adjusted to add new items, along with the existing ones:
- Top Accounts
- Verify Contract

should also take in consideration Wallet Connection and mobile view.

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #617

## Demo

https://github.com/aeternity/aescan/assets/15363559/bedb1ed2-43e2-4a69-9d8a-02add4cf1d7c

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [X]  change menu structure

### CODE REVIEW

**Comment by janmichek** [comment]
@michele-franchi I would like to ask you to evaluate the progress from a product/user perspective. No need to look at code now

**Comment by michele-franchi** [comment]
@janmichek
The base looks good but I would adjust the following:
- Blockchain menu should also contain "State Channels" and "Hyperchains" (grey color with coming soon)
- Font size should be 16px, I see it's currently 20px which feels too big
- I would add the usual dropdown menu icon to indicate its presence:
Close state:
image

Open state:
image
And also applied to the mobile version of the menu, for example:
image

**Comment by janmichek** [comment]
> Blockchain menu should also contain "State Channels" and "Hyperchains" (grey color with coming soon)

Yess, State changes are missing. Will add it. Regarding to 'comming soon'. THe functionality is preserved from previous state (tooltip on hover) I can redo it into text one if you wish so


> Font size should be 16px, I see it's currently 20px which feels too big

I decided to enlarge it as now we have more room. For desktop I am fine to change it into smaller. For mobile devices I suggest to keep it larger for the comfort of tapping on it.


> I would add the usual dropdown menu icon to indicate its presence:

Agree, will add those carets. It was on my mind, but didn't want to introduce it without consultation. Now it's clear.

**Review by Liubov-crypto — CHANGES_REQUESTED** [review]
https://github.com/aeternity/aescan/assets/69896204/cbd64471-68d5-4e4d-af0f-413490c6b9ff

**Comment by janmichek** [comment]
> 1. In Blockchain menu are still missing State Channels and Hyperchains tabs.
> 
>     2. If I'm making active AEX9 Tokens tab then Top Accounts tab will be also active (red) witch is not correct. Reproducible on mobile and web:
>        https://github.com/aeternity/aescan/assets/69896204/51c8c375-1f83-405d-b8bd-7e61f480cd84
> 
> 
> @janmichek please check it.

Thank you! fixed

**Review by Liubov-crypto — APPROVED** [review]
Fixed, LGTM. 

I have one question: I see that there is no option to close drop down menu to click on it. I have to open new drop down to close previous one. Reproducible on mobile and web. @janmichek  Is it correct behaviour? 


https://github.com/aeternity/aescan/assets/69896204/db86a938-6b7a-40d0-9463-e3c202dfdb41

**Comment by janmichek** [comment]
> Fixed, LGTM.
> 
> I have one question: I see that there is no option to close drop down menu to click on it. I have to open new drop down to close previous one. Reproducible on mobile and web. @janmichek Is it correct behaviour?
> 2024-01-09.12.36.15.mov

I would leave it for now. On the desktop it opens on hover, so I wouldn't personally have idea of closing it by clicking

**Review by michele-franchi — CHANGES_REQUESTED** [review]
I suggested some UI changes which in my opinion makes the menu look even better. 

I found some issues:
- If you switch to mobile menu on Desktop items open just with hover and not with click which becomes pretty unusable:

https://github.com/aeternity/aescan/assets/11598532/00b37150-eda3-494f-8ee3-30f0161ec472
- If on Desktop you switch on Mobile, open the menu and switch again to desktop view you will get the following:
image

**Comment by janmichek** [comment]
Thanks for your collaboration. Will adjust

**Comment by janmichek** [comment]
All fixed

**Review by michele-franchi — CHANGES_REQUESTED** [review]
- Desktop dropdown menus do not open with hover.
- Mobile dropdown does not close when clicking again.

**Comment by janmichek** [comment]
> * Desktop dropdown menus do not open with hover.
> 
>     * Mobile dropdown does not close when clicking again.

Thanks for the feedback. Not sure where I made a mistake.

File: src/components/TheNavigation.vue [hunk 1]
```diff
@@ -1,113 +1,160 @@
...
-      text-decoration: underline;
-    }
+onBeforeUnmount(() => {
+  window.removeEventListener('resize', updateIsDesktopResolution)
+})
+
+const isDesktopResolution = ref(null)
+
+function updateIsDesktopResolution() {
+  isDesktopResolution.value = isDesktop()
```
Comment (janmichek) [inline src/components/TheNavigation.vue:134]: Will follow up with this
https://github.com/aeternity/aescan/issues/105
We will reuse it here and also another places

**Comment by janmichek** [comment]
@Liubov-crypto please better check again, I made more changes

**Comment by janmichek** [comment]
@michele-franchi I am not super-satisfied with your latest suggestion about this

image

In my opinion, there should be a more visually distinguished hierarchy of different menu levels.

**Review by Liubov-crypto — APPROVED** [review]
LGTM

File: src/components/TheNavigation.vue [hunk 2]
```diff
@@ -1,113 +1,160 @@
...
+      isDisabled: true,
+    },
+  ],
+},
+{
+  name: 'Developers',
+  isActive: false,
+  submenu: [
+    {
+      name: 'Contract Verification',
```
Comment (michele-franchi) [inline src/components/TheNavigation.vue:91]:
```suggestion
      name: 'Smart Contract Verification',
```

File: src/components/TheNavigation.vue [hunk 3]
```diff
@@ -1,113 +1,160 @@
...
+    },
+    {
+      name: 'AEX141 Tokens (NFTs)',
+      path: '/nfts',
+    },
+    {
+      name: 'DEX Trades',
+      path: '/trades',
+      isDisabled: true,
+    },
```
Comment (michele-franchi) [inline src/components/TheNavigation.vue:67]:
```suggestion
```

I would remove it for now

File: src/components/TheNavigation.vue [hunk 4]
```diff
@@ -1,113 +1,160 @@
...
+      path: '/top-accounts',
+      isDisabled: true,
+    },
+    {
+      name: 'Charts',
+      path: '/charts',
+      isDisabled: true,
+    },
+  ],
+},
```
Comment (michele-franchi) [inline src/components/TheNavigation.vue:85]:
```suggestion
```

I would remove it for now.

File: src/components/MenuItem.vue [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      border-radius: 8px;
+      border: 1px solid var(--color-midnight-15);
+      box-shadow: 0 6px 30px #0000001a;
+      padding: var(--space-0);
+      min-width: 220px;
+    }
+  }
+
+  &__item {
+    padding: var(--space-0) var(--space-3);
```
Comment (michele-franchi) [inline src/components/MenuItem.vue:91]:
```suggestion
    padding: var(--space-0) var(--space-3) var(--space-0) var(--space-4);
```

I would go this way so this also solves the mobile issue where items appear at the same level:
image

**Review by michele-franchi — APPROVED** [review]
I added some more suggestions, can be merged once fixed 🚀

**Pull request merged**